docs/source/usage.rst

```diff
@@ -12,7 +15,206 @@ To use STAC Task, first install it using pip:
 
    (.venv) $ pip install stactask
 
-Doing something
-----------------
 
-tbd
+A simple task definition
+------------------------
+
+The intended use of stactask is for a developer to extend the `stactask.Task` class. An trivial
+example class that does is shown below, which accepts a payload containing a field `item_id` and
+generates a STAC Item with this id.
+
+.. code-block:: python
+
+   #!/usr/bin/env python
+
+   import logging
+   import os
+   from datetime import datetime, timezone
+   from typing import Any
+
+   from pystac import Item
+   from stactask.exceptions import InvalidInput
+   from stactask.task import Task
+
+
+   class MyTask(Task):  # type: ignore
+      name = "my-task"
+      description = "Create STAC Items for payload"
+      version = "v2024.02.01"
+
+      # override from Task
+      @classmethod
+      def validate(cls, payload: dict[str, Any]) -> bool:
+         if "item_id" not in payload:
+               raise InvalidInput("Missing field 'item_id' in payload")
+         return True
+
+      # override from Task
+      def process(self, **kwargs: Any) -> list[dict[str, Any]]:
+         item = Item(
+               id=self._payload["item_id"],
+               geometry={
+                  "type": "Polygon",
+                  "coordinates": [
+                     [
+                           [100.0, 0.0],
+                           [101.0, 0.0],
+                           [101.0, 1.0],
+                           [100.0, 1.0],
+                           [100.0, 0.0],
+                     ]
+                  ],
+               },
+               bbox=None,
+               datetime=datetime.now(timezone.utc),
+               properties={},
+         )
+
+         return [item.to_dict()]
+
+   # Support for running as a Lambda Function
+   def handler(event: dict[str, Any], context: Any) -> dict[str, Any]:
+      return MyTask.handler(event)  # type: ignore
+
+   # Support for running as a CLI application
+   if __name__ == "__main__":
+      MyTask.cli()
+
+The expected input looks like the following:
```

Comment:
Consider adding a link to ProcessPayload definition, perhaps also identifying what is important and what's not important for local development. I'm not sure the best place for this, but maybe some discussion around
- upload options are only needed if you're uploading to a stac api
- this regex matches all ids and sends it to `my-collection`, and should be used as a default, however you may want to filter specific IDs and send to different collections

Reply:
Ack, I just saw lines 99-101. This is probably OBE.